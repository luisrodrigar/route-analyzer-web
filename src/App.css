.App {
  text-align: center;
  width:100%;
  height:100%;
}

.App-header {
  background-color: #222;
  height: 40px;
  padding: 8px;
  color: white;
  margin:0;
}

.App-title {
  font-size: 1.2em;
}

.App-intro {
  font-size: large;
}

.loading-div{
	z-index:1000;
    position:fixed; 
    width:100%; 
    height:100%;
    background:rgba( 255, 255, 255, .8);
}