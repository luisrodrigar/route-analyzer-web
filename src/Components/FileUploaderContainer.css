.FileUploaderContainer{
	padding:15px;
}

.FileUploaderForm select, .FileUploaderForm input{
	padding:10px;
}

.FileUploaderForm input[type="submit"]{
	padding:10px;
}