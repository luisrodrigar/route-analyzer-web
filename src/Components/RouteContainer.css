.RouteTable{
	float:left;
	padding:10px;
}

.RouteMap{
	float:right;
	padding:20px;
}