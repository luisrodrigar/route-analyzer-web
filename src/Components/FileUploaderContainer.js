import React, { Component } from 'react';
import {upload} from '../Services/file';
import './FileUploaderContainer.css';


class FileUploaderContainer extends Component {

  constructor(props){
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.showProgressIcon();
    const formData = new FormData();
    formData.append("name", "name de ejemplo");
    formData.append('file', this.fileInput.files[0]);
    formData.append('type', this.type.value);
    
    upload(formData)
      .then(idActivity => {
        this.props.showMapRoute(idActivity);
      })
      .catch(err => {
        alert(err.message);
      });

  };

  render() {
    return (
      <div className="FileUploaderContainer">
      <form className="FileUploaderForm" onSubmit={this.handleSubmit}>
        <input 
          type="file" 
          ref={input => { this.fileInput = input; }} />
        <select defaultValue="gpx" className="optionFileUploaderForm" ref={option => { this.type = option; }} >
          <option value='gpx'>gpx</option>
          <option value='tcx'>tcx</option>
        </select>
        <input type="submit" value="Submit" />
      </form>
      </div>
    );
  }
}

export default FileUploaderContainer;