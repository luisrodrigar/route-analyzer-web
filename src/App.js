import React, { Component } from 'react';
import {BrowserRouter as Router } from 'react-router-dom'
import FileUploaderContainer from './Components/FileUploaderContainer';
import RouteContainer from './Components/RouteContainer';
import { withStyles } from 'material-ui/styles';
import { CircularProgress } from 'material-ui/Progress';
import { connect } from "react-redux";
import blue from 'material-ui/colors/blue';
import './App.css';

const styles = theme => ({
  progress: {
    margin: theme.spacing.unit * 2,
  },
});

const mapStateToProps = state => {
  return { 
    showRoute: state.app.showRoute,
    progress: state.app.progress,
    id: state.app.id
  };
};

export class App extends Component {

  render() {
    return (
      <Router>
      <div className="App">
        {this.props.progress && 
            <div className={"loading-div"}>
            <CircularProgress style={{ 
                                color: blue[600],
                                }}
                              size={240}
            />
            </div>
        }
        <header className="App-header">
          <h1 className="App-title">Route Analyzer</h1>
        </header>
        <div className="App-intro">
          <FileUploaderContainer/>
          {this.props.showRoute && <RouteContainer id={this.props.id}/>}
        </div>
      </div>
      </Router>
    );
  }
}

export default connect(mapStateToProps)(withStyles(styles)(App));
