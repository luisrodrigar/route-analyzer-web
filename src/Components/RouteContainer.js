import React, { Component } from 'react';
import {get, removePoint, removeLaps} from '../Services/activity';
import DownloadFileComponent from './DownloadFileComponent';
import {RouteMapContainer} from './RouteMapContainer';
import {getLapsTrackPoints} from '../Utils/lapOperations';
import Grid from 'material-ui/Grid';
import RouteTable from './RouteTable';
import './RouteContainer.css';

class RouteContainer extends Component {

	constructor(props){
		super(props);
		this.setActivityObject = this.setActivityObject.bind(this);
		this.removePosition = this.removePosition.bind(this);
		this.removeLaps = this.removeLaps.bind(this);
		this.state = {activity: {}, laps: []};
		this.setActivityObject(this.props.id);
	}

	componentWillReceiveProps(nextProps){
		if(nextProps.id !== this.props.id){
			this.setActivityObject(nextProps.id);
	    }
	}

	setActivityObject(id){
		get(id)
			.then( activityObject => {
				this.setState({
					activity: activityObject,
					laps: getLapsTrackPoints(activityObject.laps)
				});
			})
			.catch( err => {
				alert(err.message);
			});
	}

	removePosition(position, date, index){
    	removePoint(this.props.id,position, date, index)
	      .then( activityObject => {
	        this.setState({
	        	activity: activityObject,
	        	laps: getLapsTrackPoints(activityObject.laps)
	        });
	      })
	      .catch( err => {
	        alert(err.message);
	      });
  	}

  	removeLaps(dataSelected){
  		removeLaps(this.props.id, dataSelected)
  		.then(activityObject => {
	        this.setState({
	        	activity: activityObject,
	        	laps: getLapsTrackPoints(activityObject.laps)
	        });
	      })
	      .catch( err => {
	        alert(err.message);
	      });
  	}

	render(){
		if(this.state.activity && this.state.activity.laps){
			return (
				<div>
					<Grid container spacing={24}>
						<Grid item xs={12}>
							<DownloadFileComponent 	id={this.state.activity.id} 
													type={this.state.activity.sourceXmlType} />
						</Grid>
						<Grid item xs={7}>
							<div className="RouteTable">
								<RouteTable handleRemoveLaps={this.removeLaps} laps={this.state.laps}/>
							</div>
						</Grid>
						<Grid item xs={5}>
							<div className="RouteMap">
								<RouteMapContainer handleRemoveMarker={this.removePosition} laps={this.state.laps} />
							</div>
						</Grid>
					</Grid>
				</div>
			);
		} else
			return(
				<div>
					
				</div>
			);
	}

}

export default RouteContainer;