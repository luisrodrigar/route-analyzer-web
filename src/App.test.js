import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from './App';
import { CircularProgress } from 'material-ui/Progress';
import FileUploaderContainer from './Components/FileUploaderContainer';
import RouteContainer from './Components/RouteContainer';

describe('App main component', () => {
  it('Initial render without crashing', () => {
      const props = {
          showRoute: false,
          progress: false,
          id: null
      };
      const wrapper = shallow(<App {...props} />);
      expect(wrapper.find(CircularProgress).length).toBe(0);
      expect(wrapper.find(FileUploaderContainer).length).toBe(1);
      expect(wrapper.find(RouteContainer).length).toBe(0);
      expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Loading some activity render', () => {
        const props = {
            showRoute: false,
            progress: true,
            id: null
        };
        const wrapper = shallow(<App {...props} />);
        expect(wrapper.find(CircularProgress).length).toBe(1);
        expect(wrapper.find(FileUploaderContainer).length).toBe(1);
        expect(wrapper.find(RouteContainer).length).toBe(0);
  });
  it('Some activity loaded render', () => {
        const idActivity = "1234567890";
        const props = {
            showRoute: true,
            progress: false,
            id: idActivity
        };
        const wrapper = shallow(<App {...props} />);
        expect(wrapper.find(CircularProgress).length).toBe(0);
        expect(wrapper.find(FileUploaderContainer).length).toBe(1);
        const routeContainer = wrapper.find(RouteContainer);
        expect(routeContainer.length).toBe(1);
        expect(routeContainer.props().id).toBe(idActivity);
  });
});
