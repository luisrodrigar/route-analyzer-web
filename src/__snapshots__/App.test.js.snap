// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App main component Initial render without crashing 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1
        className="App-title"
      >
        Route Analyzer
      </h1>
    </header>
    <div
      className="App-intro"
    >
      <Connect(FileUploaderContainer) />
    </div>
  </div>
</BrowserRouter>
`;
